src/AppInstallerCommonCore/Public/AppInstallerStrings.h

```diff
@@ -230,4 +228,26 @@ namespace AppInstaller::Utility
 
     // Converts the given hexadecimal string into bytes.
     std::vector<uint8_t> ParseFromHexString(const std::string& value, size_t byteCount = 0);
+
+    // Join a string vector using the provided separator.
+    LocIndString Join(LocIndView separator, const std::vector<LocIndString>& vector);
+
+    // Superset of std::to_string supporting string convertibles as input.
+    template <typename T>
+    std::string ToString(T value)
+    {
+        std::ostringstream result;
+        result << value;
+        return result.str();
+    }
+
+    // Format an input string by replacing placeholders {index} with provided values at corresponding indices.
+    // Note: After upgrading to C++20, this function should be deprecated in favor of std::format.
+    template <typename ... T>
+    std::string Format(std::string inputStr, T ... args)
+    {
+        int index = 0;
+        (FindAndReplace(inputStr, "{" + std::to_string(index++) + "}", ToString(args)), ...);
+        return inputStr;

```

Comment:
We can scan the string after the one liner, but if we do, I would consider implementing the scan and replace placeholders approach instead since we can do all checks in one run.